cli/generate.go

```diff
@@ -168,19 +189,45 @@ func generateMake() cli.Command {
 				return errors.Wrapf(err, "generating evergreen config from build file(s)")
 			}
 
-			output, err := json.MarshalIndent(conf, "", "\t")
-			if err != nil {
-				return errors.Wrap(err, "marshalling evergreen config as JSON")
-			}
-			if outputFile != "" {
-				if err := ioutil.WriteFile(outputFile, output, 0644); err != nil {
-					return errors.Wrapf(err, "writing JSON config to file '%s'", outputFile)
-				}
-			} else {
-				fmt.Println(string(output))
+			if err := formatAndOutputGeneratedConfig(conf, c.String(outputFormatFlagName), outputFile); err != nil {
+				return errors.Wrap(err, "formatting and writing output")
 			}
 
 			return nil
 		},
 	}
 }
+
+func formatAndOutputGeneratedConfig(conf *shrub.Configuration, format, file string) error {
+	doFormatting, ok := generatedConfigFormats()[format]
+	if !ok {
+		return errors.Errorf("unrecognized output format '%s'", format)
+	}
+	output, err := doFormatting(conf)
+	if err != nil {
+		return errors.Wrapf(err, "formatting configuration as '%s'", format)
+	}
+
+	if file == "" {
+		fmt.Println(string(output))
+		return nil
+	}
+
+	if err := ioutil.WriteFile(file, output, 0644); err != nil {
+		return errors.Wrapf(err, "writing formatted config to file '%s'", file)
+	}
+
+	return nil
+}
+
+func checkGeneratedConfigFormat(c *cli.Context) error {
+	format := strings.ToLower(c.String(outputFormatFlagName))
+	var formats []string
+	for f := range generatedConfigFormats() {
+		if format == f {
+			return nil
+		}
+		formats = append(formats, format)
```

Comment:
This seems unnecessary, can't we just do a lookup on the hashmap for the format we are trying to validate? And if we really want to return the acceptable formats upon error, can't we just have slice somewhere with the list of acceptable formats?

Reply:
Fixed.